cookbooks/aws-parallelcluster-install/recipes/c_states.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster-install
+# Recipe:: c_states
+#
+# Copyright:: 2013-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+return unless node['kernel']['machine'] == 'x86_64'
+
+regen_grub_command = value_for_platform(
+  %w(centos amazon) => { 'default' => '/usr/sbin/grub2-mkconfig -o /boot/grub2/grub.cfg' },
+  'ubuntu' => { 'default' => '/usr/sbin/update-grub' }
+)
+
+ruby_block 'Append GRUB_CMDLINE_LINUX_DEFAULT if it do not exist in /etc/default/grub' do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    command = 'grep "^GRUB_CMDLINE_LINUX_DEFAULT=" /etc/default/grub'
+    command_out = shell_out(command).stdout
+
+    if command_out.include? "GRUB_CMDLINE_LINUX_DEFAULT="
+      Chef::Log.info('Found GRUB_CMDLINE_LINUX_DEFAULT line')
+    else
+      Chef::Log.warn('GRUB_CMDLINE_LINUX_DEFAULT not found - Adding')
+      command = 'echo \'GRUB_CMDLINE_LINUX_DEFAULT=""\' >> /etc/default/grub'
+      shell_out(command)
+      Chef::Log.info('Added GRUB_CMDLINE_LINUX_DEFAULT line')
+    end
+  end
+  action :run
+end
+
+ruby_block 'Add "intel_idle.max_cstate=1" to /etc/default/grub in line GRUB_CMDLINE_LINUX_DEFAULT' do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    command = 'grep "^GRUB_CMDLINE_LINUX_DEFAULT=" /etc/default/grub'
+    command_out = shell_out(command).stdout
+
+    if command_out.include? "intel_idle.max_cstate"
+      Chef::Log.info('Found intel_idle.max_cstate in GRUB_CMDLINE_LINUX_DEFAULT - I do not change value:' + command_out)
```

Comment:
I do not change value: --> "Leaving value:" --> or just "value:"

Reply:
Done